Formula/datree.rb

```diff
@@ -14,7 +14,7 @@ class Datree < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "0fca46a8f5d4d01ec05ab2b9ec942e8e7e72d7581104b7de51c1a7390c5e91aa"
   end
 
-  depends_on "go" => :build
+  depends_on "go@1.16" => :build
```

Comment:
Why?

Reply:
When installing datree using `brew install datree` and then running `datree test`, I encounter this error:
```
fatal error: unexpected signal during runtime execution
```
but using `go@1.16` datree works fine.
I'm still trying to understand this issue's source (I suspect it is related to a specific `x/sys` version) but until then I would like to pin the go version so users will be able to use it.

Reply:
Hi @carlocab,
At the moment running `brew install datree` downloads datree succesfully but running `datree` results in panic. `datree` is not building well with 1.17 version.
Link to the issue: https://github.com/datreeio/datree/issues/159

This is also similar to #84487

We need to fix this because this is affecting all macos users

Reply:
Well, it seems to be all macOS users -1 because I can't reproduce the issue when running the same commands as in the issue.

Reply:
You're right. We can't find the exact reason why this is not working for our users. Can we push this fix in the meanwhile until we find a solution for this issue?

Reply:
at least for me, `brew test datree` passes but directly running `datree version` produces the stacktrace

Reply:
to follow up here, we did not catch this error because it does not produce with `TERM=dumb`, which our test environment sets